lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -281,17 +281,15 @@ def _get_pubkey_content():
             else:
                 return False
 
-        def _parse_pubkey():
-            pubkey_content = _get_pubkey_content()
+        def _parse_pubkey(pubkey_content):
             if pubkey_content:
                 parts = pubkey_content.split(' ', 2)
                 return parts[0], parts[1], '' if len(parts) <= 2 else parts[2]
             return False
 
         def _pubkey_valid(pubkey):
             if pubkey_parts:
-                current_pubkey = ' '.join([pubkey_parts[0], pubkey_parts[1]])
-                return current_pubkey == pubkey
+                return pubkey_parts[:2] == _parse_pubkey(pubkey)[:2]
```

Comment:
Is it not weird that this function operates on a outer-scope variable (``pubkey_parts``) that may not be defined? It's only defined just before the first call to ``_pubkey_valid`` on line 307. If you're going to use outer-scope variables you might as well also just call ``_get_pubkey_content()`` instead of passing in ``pubkey``.

Reply:
I didn't come up with this design. Similar things are done (or at least have been done in the past) in multiple crypto modules. I don't mind changing this, but I don't want to do it in this bugfix PR. Is that ok for you?

Reply:
Seems fair.

Reply:
Unclear how I can dismiss this review...

Reply:
Unclear how I can dismiss this review...

Maybe this works.

Reply:
Unclear how I can dismiss this review...

Maybe this works.

Reply:
You need to add a new review with "Approved". AFAIK that's the only way to get rid of an existing "Request changes" (I also had enough "fun" trying to figure that out in the past...).